### DISCUSSION TITLE
Technical explanation of custom themes

### DISCUSSION THREAD

**Discussion by GomezIvann** [dxc-technology/halstack-react#1354] [General] [answered]
We need to have an explanation of how to use the `HalstackProvider` with the different theming strategies we offer on the new site. The topics to discuss are:

**1. Where should we locate this section?
2. What content should it have?**

[Older site approach](https://developer.dxc.com/tools/react/next/#/overview), section _"Custom Themes"_.

**Comment by Jialecl** [comment]
Right now there is an example and a small paragraph regarding the  `HalstackProvider` on the Localization page.
I would consider the same approach for theming.

There is another option which is having a page for the  `HalstackProvider` and linking to the Theme and Localization Page, but I prefer the first option.

**Reply by eriesgo** [reply]
In the long term, I want Halstack to trial the four quadrants approach for documentation. 
https://www.thoughtworks.com/radar/techniques/documentation-quadrants

We have done a good job with the Explanation and Reference part, but we have room for improvement in Tutorials and How-to-guides. Custom themes could be a good topic to create a how-to guide like **How to theme an application**. In how-guides we focus in the problem to be solved (theming) not in the solution (``HalstackProvider``)

**Reply by GomezIvann** [reply]
We need to think and discuss how can we integrate more didactic sections into our documentation site, completely agree with you on this Enrique.

**Comment by GomezIvann** [comment, marked as answer]
We have already added a new Halstack Provider section alongside a more complete and didactic section of Theming:

- https://developer.dxc.com/halstack/next/principles/themes/
- https://developer.dxc.com/halstack/next/utilities/halstack-provider/
